### PR TITLE
Clarification of NextDNS logging policy

### PR DESCRIPTION
**Pull request by xe-3**
Changes proposed in this PR:

- CONTEXT: This PR revises footnote 5 in the DNS section (NextDNS' logging policy) it builds upon and further clarifies 67614c3 and is an alternative to the proposed `PR 2427`.

- PROBLEM: The current description needs clarification and seemingly contradicts itself (it begins by stating logging is opt-out and ends by saying no logs are kept without the user 'specifically requesting it'). It also does not differentiate NextDNS's public DNS servers `dns.nextdns.io` from the personal `my.nextdns.io/<ID>`.

- There is a pending PR (2427) that proposes to amend that footnote, but the proposed changes in that PR leave out important information that can help readers make an informed choice. My assessment is that PR 2427 helpfully clarifies one inaccuracy but introduces its own inaccuracy by omission. The author of that PR has so far been unwilling to amend or modify it. So I am creating this PR as an alternative.

- GOAL: The goal of this PR is to provide readers clearer and fuller information than the current footnote provides, and provide fuller information than the other proposed pull request (2427) and use neutral language.

- SPECIFIC CHANGES: This PR is an improvement in three specific ways, (1) it clarifies the seemingly contradictory statements about logging (2) it explains to readers how the the logging policy will apply to NextDNS's public and personal DNS services, and (3) it alerts readers to the apparent contradiction between the NextDNS privacy policy, and the default account settings.

- The proposed language is:
>When used with a user account, NextDNS will enable insights and logging features by default (note: this seemingly conflicts with their privacy policy). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)

- IF this PR is accepted [PR 2427](https://github.com/privacyguides/privacyguides.org/pull/2427) should be rejected. IF 2427 is accepted this PR should be rejected.

-----------------------------------

- [x] I have disclosed any relevant conflicts of interest in my post.
- [x] I agree to grant Privacy Guides a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform, relicense, and distribute my contribution as part of this project.
- [x] I am the sole author of this work. 
- [x] I agree to the [Community Code of Conduct](https://www.privacyguides.org/en/code_of_conduct/).

### CODE REVIEW

**Comment by dngray** [comment]
> IF this PR is accepted [PR 2427](https://github.com/privacyguides/privacyguides.org/pull/2427) should be rejected. IF 2427 is accepted this PR should be rejected.

I was looking at that and felt that it didn't really make things clearer.

> logging features by default (note: this seemingly conflicts with their privacy policy).

I think we could make this sound less negative. The privacy policy relates to logging they would do for potentially other purposes like external reporting.

Obviously if you enable a logging feature because you want insight on your activity that has to be stored somewhere. I think it's unfair to say this is "contrary to privacy policy".

**Comment by archerallstars** [comment]
This PR opens the footnote with:

> When used with a user account,

I think it's worth mentioning that on https://nextdns.io, the **"Try it now"** buttons at both the top and the bottom are described with "No signup required." With this, the users _might_ assume that _their account_ has not been created/used, since they didn't sign up for the service. 

Moreover, there's no public DNS address (`dns.nextdns.io`) mentioning on the website at all. Even the mockup screenshot on the website that shows an example DNS address, is an address with the user account:

image

I doubt using the service without a user account is intended by the team at NextDNS at all.

@dngray 

>> logging features by default (note: this seemingly conflicts with their privacy policy).

> I think we could make this sound less negative. The privacy policy relates to logging they would do for potentially other purposes like external reporting.

> Obviously if you enable a logging feature because you want insight on your activity that has to be stored somewhere. I think it's unfair to say this is "contrary to privacy policy".

The problem with privacy policy is not about the purpose of the log or the way the log is used, but the default behavior of the logging that's supposed to be **_opt-in_**, not _**opt-out**_. This is even more problematic considering that the logging is _optional_, regardless of whether the user has an account.

**Comment by dngray** [comment]
> "seemingly" is almost misleading

I don't particularly like that word either, it makes it sound as if this is the fault of Next DNS. To me it just seems to be whether or not the user wants to make use of a feature that requires the logging to operate.

I don't think this is the same as a opaque privacy policy where it might not be clear for whatever purpose.

For example with something like GDPR wouldn't this come under "legal basis" being that the feature needs it to work?

File: docs/dns.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Encrypted DNS with third-party servers should only be used to get around basic [
...
 [^3]: Control D only logs for Premium resolvers with custom DNS profiles. Free resolvers do not log data. [https://controld.com/privacy](https://controld.com/privacy)
 [^4]: Mullvad's DNS service is available to both subscribers and non-subscribers of Mullvad VPN. Their privacy policy explicitly claims they do not log DNS requests in any way. [https://mullvad.net/en/help/no-logging-data-policy/](https://mullvad.net/en/help/no-logging-data-policy/)
-[^5]: NextDNS can provide insights and logging features on an opt-out basis. You can choose retention times and log storage locations for any logs you choose to keep. If it's not specifically requested, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
+[^5]: When used with a user account, NextDNS will enable insights and logging features by default (note: this seemingly conflicts with their privacy policy). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
```
Comment (dngray) [inline docs/dns.md:27]:
```suggestion
[^5]: When used with a user account, NextDNS will enable insights and logging features by default as the feature requires it. You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
```

**Comment by xe-3** [comment]
> I was looking at that and felt that it didn't really make things clearer.

Fair enough. Is there something specific that you feel is not clear? If it's just the note about the privacy policy, I would be willing to remove that.

The primary point I wanted to clarify was that:
-  _With an account_, logs are enabled by default (but can be disabled or configured to a user's preferences).
-  _Without an account_ NextDNS keeps no logs. 

> I think we could make this sound less negative.

I agree. I struggled to find language that was both accurate, and neutral. I couldn't find a way to accomplish both so I chose accuracy. But I don't want to mislead readers into thinking logs are inherently bad, in the context of NextDNS logging is a feature not a flaw (but still a risk for some threat models)

How would you rephrase it? Or is your preference that the note on conflicting with the privacy policy be removed?

> The privacy policy relates to logging they would do for potentially other purposes like external reporting.
> Obviously if **you enable** a logging feature because you want insight on your activity that has to be stored somewhere. I think it's unfair to say this is "contrary to privacy policy"

On this point, I agree with @archerallstars:
> The problem with NextDNS privacy policy is not about the purpose of the log or the way the log is used, but the default behavior of the logging that's supposed to be opt-in, not opt-out.

I suppose it isn't strictly necessary to explicitly note that conflict. As long as we make it clear that when you signup for an account logs are enabled by default (and I believe this PR does that, with or without the note on the privacy policy).

File: docs/dns.md (hunk 1 above)
Comment (dngray) [reply docs/dns.md:27]: I'm thinking something like this, that clearly states the "logging/insights" feature require some "logging".

Comment (xe-3) [reply docs/dns.md:27]: @dngray I'm onboard with your amendment. I'm editing it now.

Comment (xe-3) [reply docs/dns.md]: Okay, the PR has been updated. New language:

> When used with a user account, NextDNS will enable insights and logging features by default (as some features require it). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)

Comment (archerallstars) [reply docs/dns.md]: @dngray 

> For example with something like GDPR wouldn't this come under "legal basis" being that the feature needs it to work?

Even if the service **needs** the logging to work, I believe that the privacy policy should state it clearly. But currently, it's stated in the opposite direction. 

Moreover, as a DNS resolver service, usage insight is not a hard requirement, e.g. resolving DNS queries. The logging is optional at best.

**Comment by archerallstars** [comment]
@dngray

> For example with something like GDPR wouldn't this come under "legal basis" being that the feature needs it to work?

Even if the service needs the logging to work, I believe that the privacy policy should state it clearly. But currently, it's stated in the opposite direction.

Moreover, as a DNS resolver service, usage insight is not a hard requirement, e.g. resolving DNS queries. The logging is optional at best.

**Comment by archerallstars** [comment]
@xe-3 

> As long as we make it clear that when you signup for an account logs are enabled by default

Even without the user consent to sign up for an account, the logs are also enabled by default. Here is the button on the website:

image

It says: **"No signup required."**

**Comment by xe-3** [comment]
The button in your screenshot is a ~4 day free trial account (if you click the button, you'll see a big blue banner across the top that indicates that, and tells you you'll need to signup once trial period is over).

As to "no signup required" I agree with you. I may have accidentally used the term 'signup' in these comments somewhere, but I intentionally phrased the PR to avoid the term "signup" so as not to cause confusion.

The language I've used is:
> "When used with a user account"

**Comment by dngray** [comment]
I think whats there now is good though.

Perhaps someone should raise with NextDNS the issue about the privacy policy not being entirely clear?

**Comment by xe-3** [comment]
> Perhaps someone should raise with NextDNS the issue about the privacy policy not being entirely clear?

I've inquired about the privacy policy. I haven't heard back from them yet. When I do here back, I'll post an update on the forum about it.

**Review by dngray — APPROVED** [review]

**Review by rollsicecream — APPROVED** [review]

File: docs/dns.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ Encrypted DNS with third-party servers should only be used to get around basic [
...
 [^3]: Control D only logs for Premium resolvers with custom DNS profiles. Free resolvers do not log data. [https://controld.com/privacy](https://controld.com/privacy)
 [^4]: Mullvad's DNS service is available to both subscribers and non-subscribers of Mullvad VPN. Their privacy policy explicitly claims they do not log DNS requests in any way. [https://mullvad.net/en/help/no-logging-data-policy/](https://mullvad.net/en/help/no-logging-data-policy/)
-[^5]: NextDNS can provide insights and logging features on an opt-out basis. You can choose retention times and log storage locations for any logs you choose to keep. If it's not specifically requested, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
+[^5]: When used with a user account, NextDNS will enable insights and logging features by default (as some features require it). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
```
Comment (jonaharagon) [inline docs/dns.md:27]:
```suggestion
[^5]: When used with an account, NextDNS will enable insights and logging features by default (as some features require it). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without an account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
```

Comment (archerallstars) [reply docs/dns.md:27]: Are you sure NextDNS privacy policy number 3 refer to the usage without an account (as the logging can only be requested on an user account)?

We are not trying to cover something, are we? The service clearly violated its policy, why do we still act like nothing happened? An the last question, if this was not NextDNS, do we still recommend the service/app that violated its policy?

**Pull request merged**